lib/mapper_tools.py

```diff
@@ -85,53 +86,56 @@ def loadMapperData(mode = 'train', data_dir_train = DATA_DIR_TRAIN, data_dir_tes
     return data_train, data_header_train, data_test, data_header_test
 
 def getLatentRep(n_components, data, remake=True):
-    label = "PCA%d" % (n_components)
-    if not remake and os.path.exists("%s_latent" % (label)):
-        frin = open(TEMP_DATA+"%s_latent" % (label), "rb")
-        rep = pickle.load( frin )
+    label = "PCA%d" % n_components
+    if not remake and os.path.exists("%s_latent" % label):
+        frin = open(TEMP_DATA+"%s_latent" % label, "rb")
+        rep = pickle.load(frin)
         return rep
 
+    if log.level == logging.DEBUG:
+        print("------> fitting PCA to data of shape {}...".format(data.shape))
     pca = MyPCA(n_components=n_components)
     pca.fit(data)
 
+
     rep = LatentRep([pca], label)
-    fr = open(TEMP_DATA+"%s_latent" % (label), "wb")
+    fr = open(TEMP_DATA+"%s_latent" % label, "wb")
     pickle.dump(rep, fr)
```

Comment:
pickle Protocols are generally not backwards compatible. You may want to make a note of which Protocol you used